### ISSUE TITLE
Persist RepositoryObject in CocinaObjectStore store method

### ISSUE THREAD

**Issue by justinlittman** [sul-dlss/dor-services-app#4798]
For the CocinaObjectStore store method, implement such that when enabled_features.repository_object_store is enabled:
* If the RepositoryObject does not exist, it will be created.
* If the RepositoryObject is not open, a RepositoryObject::VersionNotOpened will be raised.
* The RepositoryObject's open RepositoryObjectVersion is updated with the provided cocina.

Store will continue to update the Dro / Collection / AdminPolicy.

**Comment by jcoyne** [comment]
@justinlittman I think this should go in CreateObjectService instead.  Currently that is where we do the validation on update.

**Comment by jcoyne** [comment]
blocked by #4788

**Comment by lwrubel** [comment]
REquires updating to align with CocinaObjectStore proposal.

**Comment by justinlittman** [comment]
Closing in favor of a new ticket.
